config/next/config.serviceWorker.js

```diff
@@ -0,0 +1,83 @@
+const withOffline = require("next-offline");
+const { PHASE_DEVELOPMENT_SERVER } = require("next/constants");
+
+module.exports = (nextConfig = {}, { nextComposePlugins, phase }) => {
+  const swDest = "static/service-worker.js";
+  const config = {
+    ...nextConfig,
+    env: {
+      ...nextConfig.env,
+      SERVICE_WORKER_EXISTS: true,
+      SERVICE_WORKER_TIMEOUT: "60000",
+    },
+    target: "serverless",
+    generateInDevMode: process.env.SERVICE_WORKER === "true",
+    sourcemap: true,
+    transformManifest: manifest => ["/"].concat(manifest),
+    workboxOpts: {
+      swDest,
+      exclude: [
+        /\.map$/,
+        /\manifest.*\.js(?:on)?$/,
+        /\.js.map$/,
+        /\.css.map/,
+        /\.xls$/,
+        /\.pdf$/,
+        /\.csv$/,
+      ],
+      /**
+       * Next does not generate index.html for pages calling getServerSideProps and getInitialProps, and probably
+       * we will use it to get initial metadata.
+       * (?) We will have to use sth like dynamicUrlToDependencies / templatedURLs.
+       * https://developers.google.com/web/tools/workbox/guides/migrations/migrate-from-sw?hl=en
+       * https://stackoverflow.com/a/42182693
+       * navigateFallback: "index.html",
+       *  navigateFallbackDenylist: [
+       *    /\/graphql/,
+       *    /\/dashboard/,
+       *    /\/media\/export_files/,
+       *    /\/plugins/,
+       *    /\/storybook/,
+       *    /\/__/, // used by cypress tests runner
+       *    /\?(.+&)?no-cache=1$/,
+       *  ],
+       *  Or NetworkOnly might have the same effect.
+       */
```

Comment:
I'll get back to this after some SSR work to validate it so I want to leave it for now.